File: C/factorial.c
#include <stdio.h>

long int factorial(int number);

int main()
{
    int number = 0;
    printf("Enter a number: ");
    scanf("%d", &number);

    printf("Factorial of %d is %ld\n", number, factorial(number));
    return 0;
}

/**
 * Calculates the factorial of a given number.
 * @param number The number for which to calculate the factorial.
 * @return The factorial of the given number.
 */
long int factorial(int number)
{

    if (number == 1 || number == 0)
    {
        return 1;
    }
    else
    {
        return number * factorial(number - 1);
    }
}

File: Python/factorial.py
def factorial(number: int) -> int:
    """
    Calculate the factorial of a given number.

    Args:
        number (int): The number for which to calculate the factorial.

    Returns:
        int: The factorial of the given number.
    """
    if number == 0 or number == 1:
        return 1
    else:
        return number * factorial(number - 1)


def main():
    number = int(input("Type a number: "))
    print(f"{number}! is {factorial(number)}")


if __name__ == "__main__":
    main()
